src/main/java/cs2103/v15_1j/jimjim/ui/BottomPaneController.java

```diff
@@ -0,0 +1,249 @@
+		commandBar.setOnKeyPressed(new EventHandler<KeyEvent>() {
+			public void handle(final KeyEvent keyEvent) {
+				if (keyEvent.getCode() == KeyCode.UP) {
+					previousCommand();
+					keyEvent.consume();
+				}
+				else if (keyEvent.getCode() == KeyCode.DOWN) {
+					nextCommand();
+					keyEvent.consume();
+				}
+			}
+		});
+
+		AnchorPane.setLeftAnchor(commandBar, BORDER_WIDTH);
+		AnchorPane.setBottomAnchor(commandBar, BORDER_WIDTH);
+		AnchorPane.setRightAnchor(commandBar, COMMAND_BAR_RIGHT_BORDER);
+	}
+
+	private void setUpExecuteBtn(){
+		executeBtn = new Button("Execute");
+		executeBtn.setPrefWidth(EXECUTE_BTN_WIDTH);
+		executeBtn.setPrefHeight(EXECUTE_BTN_HEIGHT);
+		executeBtn.setOnAction(event -> handleCommand());
+		AnchorPane.setBottomAnchor(executeBtn, BORDER_WIDTH);
+		AnchorPane.setRightAnchor(executeBtn, BORDER_WIDTH);
+	}
+
+	private void setUpHelpBtn(){
+		helpBtn = new Button("Help");
+		helpBtn.setPrefWidth(EXECUTE_BTN_WIDTH);
+		helpBtn.setPrefHeight(EXECUTE_BTN_HEIGHT);
+		helpBtn.setOnAction(event -> showHelp());
+		AnchorPane.setTopAnchor(helpBtn, BORDER_WIDTH);
+		AnchorPane.setLeftAnchor(helpBtn, BORDER_WIDTH);
+	}
+
+	private void addHelpMessage(String helpMessage){
+		Circle dot = new Circle(3.0, Color.BLUE);
+		GridPane.setHalignment(dot, HPos.CENTER);
+		helpPane.addColumn(0, dot);
+
+		Label helpLabel = new Label(helpMessage);
+		helpPane.addColumn(1, helpLabel);
+	}
+
+
+	private void previousCommand(){
+		if(commandHistoryPosition == -1){
+			tempCommand = commandBar.getText();
+		}
+
+		if((commandHistoryPosition + 1) < commandHistory.size()){
+			commandHistoryPosition++;
+			commandBar.setText(commandHistory.get(commandHistoryPosition));
+			commandBar.positionCaret(commandBar.getText().length());
+		}
+	}
+
+	private void nextCommand(){
+		if((commandHistoryPosition - 1) >= 0){
+			commandHistoryPosition--;
+			commandBar.setText(commandHistory.get(commandHistoryPosition));
+			commandBar.positionCaret(commandBar.getText().length());
+		}
+		else if (commandHistoryPosition != -1) {
+			commandHistoryPosition--;
+			commandBar.setText(tempCommand);
+			commandBar.positionCaret(commandBar.getText().length());
+		}
+
+	}
+
+	public void showNotification(String msg){
+		notificationPane.setText(msg);
+		notificationPane.show();
+
+		Task<Void> sleeper = new Task<Void>() {
+			@Override
+			protected Void call() throws Exception {
+				try {
+					Thread.sleep(notificationTimeoutLength);
+				} catch (InterruptedException e) {
+				}
+				return null;
+			}
+		};
+		sleeper.setOnSucceeded(new EventHandler<WorkerStateEvent>() {
+			@Override
+			public void handle(WorkerStateEvent event) {
+				notificationPane.hide();
+			}
+		});
+		new Thread(sleeper).start();
+	}
+
+	public void focusCommandBar(){
+		commandBar.requestFocus();
+	}
+
+	public void showHelp(){
+		helpPopOver.show(helpBtn);
+	}
+
+	private void handleCommand() {
+		if (commandBar.getText() != null) {
```

Comment:
if i press enter instead of clicking execute, a null pointer is still returned causing exception.